src/components/__tests__/InstantSearch.js

```diff
@@ -3,8 +3,15 @@ import { mount } from '@vue/test-utils';
 import instantsearch from 'instantsearch.js/es';
 import InstantSearch from '../InstantSearch';
 import { version } from '../../../package.json';
+import { warn } from '../../util/warn';
 
-beforeEach(() => jest.clearAllMocks());
+jest.mock('../../util/warn');
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  warn.mockReset();
+  warn.mockRestore();
```

Comment:
I don't think `warn.mockReset()` and `warn.mockRestore()` are useful here.

* The value of the mock's usage data are already reset by `jest.clearAllMocks()`.
* `warn.mockReset()` also removes mocked return values or implementations but there is none here.
* `warn.mockRestore()` only works for mocks created with `jest.spyOn()` which is not the case here.